src/syntax/abs/mod.rs

```diff
@@ -138,6 +139,41 @@ pub fn trans_expr_inner(
         Expr::ConsType(_) => unimplemented!(),
         Expr::Bot(ident) => Ok(Abstract::Bot(ident.info.clone())),
         Expr::Sum(_) => unimplemented!(),
-        Expr::Pi(_, _) => unimplemented!(),
+        Expr::Pi(params, result) => {
+            let mut pi_env = local_env.clone();
+            let mut pi_map = local_map.clone();
+            params
+                .iter()
+                .try_fold(
+                    Either::Left(None),
+                    |pi_abs: (Either<Option<Abstract>, Abstract>),
+                     param|
+                     -> TCM<Either<Option<Abstract>, Abstract>> {
```

Comment:
I recommend aliasing this type somewhere else

Reply:
Why am I using `Either` here??? `Option` will work